code/datums/components/jammer.dm

```diff
@@ -0,0 +1,21 @@
+/datum/component/jammer
+	var/active = FALSE
+	var/range
+	var/jammer_name
+
+/datum/component/jammer/Initialize(range, var/jammer_name = parent)
+	if(istype(parent, obj/item))
+		RegisterSignal(COMSIG_ITEM_ATTACK_SELF, .proc/Toggle)
+	src.range = range

```

Comment:
>>>SRC

Reply:
range = range

OK kevinz

Reply:
this will set jammer's range var to... jammer's range var

10/10

Reply:
src.a is the same as a whats the point of this

Reply:
The arg is called range, it should be called something else.

Reply:
hnng I remember seeing this somewhere, alright I'll fix that

Reply:
@Supermichael777 not in all circumstances, range is an arg in this var which means plain `range` will use that first. `src.` tells the code that we want the arg from the jammer instead.

Reply:
wait, I wasn't wrong for once?

Reply:
No but @JJRcop is right that you should rename the arg